pkg/kine/drivers/sqlite/sqlite.go

```diff
@@ -153,43 +148,33 @@ func setup(ctx context.Context, db *sql.DB) error {
 // migrate tries to migrate from a version of the database
 // to the target one.
 func migrate(ctx context.Context, txn *sql.Tx) error {
-	var userVersion int32
+	var currentVersion SchemaVersion
 
 	row := txn.QueryRowContext(ctx, `PRAGMA user_version`)
-	if err := row.Scan(&userVersion); err != nil {
+	if err := row.Scan(&currentVersion); err != nil {
 		return err
 	}
-	currentMajor, currentMinor := getMajorMinorVersion(int32(userVersion))
 
```

Comment:
We need to add a check for version compatibility here as well as someone might have upgraded in the meanwhile.